plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/util/HudiAvroSerializer.java

```
@@ -0,0 +1,290 @@
+                    type.writeLong(output, toTrinoTimestamp(((Utf8) value).toString()));
+                }
+                else if (type.equals(TIME_MICROS)) {
+                    type.writeLong(output, (long) value * PICOSECONDS_PER_MICROSECOND);
+                }
+                else {
+                    throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Unhandled type for %s: %s", javaType.getSimpleName(), type));
+                }
+            }
+            else if (javaType == double.class) {
+                type.writeDouble(output, ((Number) value).doubleValue());
+            }
+            else if (type.getJavaType() == Int128.class) {
+                writeObject(output, type, value);
+            }
+            else if (javaType == Slice.class) {
+                writeSlice(output, type, value);
+            }
+            else if (javaType == LongTimestampWithTimeZone.class) {
+                verify(type.equals(TIMESTAMP_TZ_MICROS));
+                long epochMicros = (long) value;
+                int picosOfMillis = toIntExact(floorMod(epochMicros, MICROSECONDS_PER_MILLISECOND)) * PICOSECONDS_PER_MICROSECOND;
+                type.writeObject(output, fromEpochMillisAndFraction(floorDiv(epochMicros, MICROSECONDS_PER_MILLISECOND), picosOfMillis, UTC_KEY));
+            }
+            else if (type instanceof ArrayType arrayType) {
+                writeArray((ArrayBlockBuilder) output, (List<?>) value, arrayType);
+            }
+            else if (type instanceof RowType rowType) {
+                writeRow((RowBlockBuilder) output, rowType, (GenericRecord) value);
+            }
+            else {
+                throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Unhandled type for %s: %s", javaType.getSimpleName(), type));
+            }
+        }
+        catch (ClassCastException ignore) {
+            // returns null instead of raising exception
+            output.appendNull();
```

why swallow the exception here?

---

This is copied from the Hive connector as I remember.  Will double check this again.